baseband/mark4/base.py

```diff
@@ -357,69 +356,14 @@ def _last_header(self):
         last_header.infer_decade(self.start_time)
         return last_header
 
-    def read(self, count=None, out=None):
-        """Read count samples.
-
-        The range retrieved can span multiple frames.
-
-        Parameters
-        ----------
-        count : int
-            Number of samples to read.  If omitted or negative, the whole
-            file is read.  Ignored if ``out`` is given.
-        out : `None` or array
-            Array to store the data in. If given, ``count`` will be inferred
-            from the first dimension.  The other dimension should equal
-            ``sample_shape``.
-
-        Returns
-        -------
-        out : array of float
-            The first dimension is sample-time, and the second, given by
-            ``sample_shape``, is (channel,).  Any dimension of length unity is
-            removed if ``self.squeeze=True``.
-        """
-        if out is None:
-            if count is None or count < 0:
-                count = self.size - self.offset
-                if count < 0:
-                    raise EOFError
-
-            out = np.empty((count,) + self.sample_shape,
-                           dtype=self._frame.dtype)
-        else:
-            assert out.shape[1:] == self.sample_shape, (
-                "'out' should have trailing shape {}".format(self.sample_shape))
-            count = out.shape[0]
-
-        offset0 = self.offset
-        while count > 0:
-            frame_nr, sample_offset = divmod(self.offset,
-                                             self.samples_per_frame)
-            if frame_nr != self._frame_nr:
-                self._read_frame()
-
-            # Set decoded value for invalid data.
-            self._frame.invalid_data_value = self.fill_value
-            # Determine appropriate slice to decode.
-            nsample = min(count, self.samples_per_frame - sample_offset)
-            sample = self.offset - offset0
-            data = self._frame[sample_offset:sample_offset + nsample]
-            data = self._squeeze_and_subset(data)
-            # Copy relevant data from frame into output.
-            out[sample:sample + nsample] = data
-            self.offset += nsample
-            count -= nsample
-
-        return out
-
-    def _read_frame(self):
-        frame_nr = self.offset // self.samples_per_frame
+    def _read_frame(self, frame_nr):
         self.fh_raw.seek(self.offset0 + frame_nr * self.header0.framesize)
-        self._frame = self.read_frame(ntrack=self.header0.ntrack,
-                                      ref_time=self.start_time)
-        # Convert payloads to data array.
-        self._frame_nr = frame_nr
+        frame = self.read_frame(ntrack=self.header0.ntrack,
+                                ref_time=self.start_time)
+        # Set decoded value for invalid data.
+        frame.invalid_data_value = self.fill_value
```

Comment:
For consistency with the other formats, we should add a check that the new frame's timestamp is consistent with the offset.  This might run into issue #174.

Reply:
Since there was no test, I felt I could leave it to another PR. To remind us: #176